Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -95,4 +107,21 @@
         
         return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.OK);
     }
+
+    private void addAlternativeIds(String id, ConceptEntry entry)
+            throws IllegalAccessException, LuceneException, IndexerRunningException {
+
+        // Check if this id is same as the id sent by user
+        if (!entry.getId().equalsIgnoreCase(id)) {
+            entry.getAlternativeIds().add(id);
+        }
+
+        // Fetch the internal concept id based on the wordnetid
+        Map<String, String> fieldMap = new HashMap<String, String>();
+        fieldMap.put(SearchFieldNames.WORDNETID, entry.getWordnetId());
+        ConceptEntry[] conceptEntries = indexService.searchForConcepts(fieldMap, "AND");
```

Comment:
As per your comments, if the user makes a search with a specific wordnet id, we should display the following in alternative ids. 

- specific wordnet id
- CP id 

1) what should I display in the <digitalHPS:id concept_id> Should I display the same specific wordnet id? 

And if the user searches with the general wordnet id

- CP id
- specific wordnet id
- general wordnet id (from request)

2) what should I display in the <digitalHPS:id concept_id> Should I display the same general wordnet id or specific wordnet id ?  

So my understanding is alternative ids with have CP id and specific wordnet ids irrespective of the scenario and general wordnet id will be added only when user queries with general wordnet id.